### PR TITLE
[layout] add missing interface to set atlas feature

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in [hunk 1]
```diff
@@ -109,6 +109,13 @@ Returns whether the atlas preview mode is enabled in the designer.
 .. seealso:: :py:func:`setAtlasPreviewEnabled`
 
 .. versionadded:: 3.4
+%End
+
+    virtual void setAtlasFeature( QgsMapLayer *layer, const QgsFeature &feature ) = 0;
```
Comment (nyalldawson) [inline python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in:114]: Can't we get the Atlas layer from the layout? At best this argument is redundant, at worst it's going to cause confusion/issues.

Comment (3nids) [reply python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in]: @nyalldawson good point, see last commit, I dropped the argument

File: python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in [hunk 2]
```diff
@@ -109,6 +109,13 @@ Returns whether the atlas preview mode is enabled in the designer.
 .. seealso:: :py:func:`setAtlasPreviewEnabled`
 
 .. versionadded:: 3.4
+%End
+
+    virtual void setAtlasFeature( const QgsFeature &feature ) = 0;
+%Docstring
+Sets the specified feature as the current atlas feature
```
Comment (nyalldawson) [inline python/gui/auto_generated/layout/qgslayoutdesignerinterface.sip.in:116]: \warning it is the caller's responsibility to ensure that \a feature is a feature from the layout's current atlas coverage layer.

**Comment by nyalldawson** [comment]
Thanks!

**Review by nyalldawson — APPROVED** [review]

File: src/app/layout/qgslayoutdesignerdialog.h [hunk 3]
```diff
@@ -177,8 +178,15 @@ class QgsLayoutDesignerDialog: public QMainWindow, public Ui::QgsLayoutDesignerB
 
     /**
      * Sets the specified feature as the current atlas feature
+     * \deprecated since QGIS 3.14 use setAltasFeature without the layer argument
      */
-    void setAtlasFeature( QgsMapLayer *layer, const QgsFeature &feat );
+    Q_DECL_DEPRECATED void setAtlasFeature( QgsMapLayer *layer, const QgsFeature &feat ) SIP_DEPRECATED;
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.h:183]: This is in app, so you can just remove

**Pull request merged**